internal/gps/source_manager.go

```diff
@@ -148,7 +148,7 @@ var _ SourceManager = &SourceMgr{}
 // gps's SourceManager is intended to be threadsafe (if it's not, please file a
 // bug!). It should be safe to reuse across concurrent solving runs, even on
 // unrelated projects.
-func NewSourceManager(cachedir string) (*SourceMgr, error) {
+func NewSourceManager(cachedir string, log func(v ...interface{})) (*SourceMgr, error) {
```

Comment:
With regard to the one-off type, I'm happy to use a `*log.Logger` instead.  My WIP branch has more usages, and at some point this seemed cleaner, but maybe it only reduced imports.

Regarding the rest, can you elaborate on what you are suggesting?
For more context, there is currently an additional `cacheAge` parameter in my WIP branch as well. Both need to be customizable for testing at the very least, so would an unexported constructor be acceptable alternative for now?

Long term, should this API take some sort of forward compatible configuration (e.g. single struct, functional options, etc)?